### ISSUE TITLE
Nightly tarball build not properly concealing help text

### ISSUE THREAD

**Issue by mikesmithgh** [neovim/neovim#27943]
### Problem

The nightly tarball build of Neovim is not properly concealing text. I am noticing the first character is concealed properly but the last character is visible in help files.

For example, given the following text with the filetype `help` and a `conceallevel` set to 1, 2, or 3.
```
`test`
|test|
*test*
```

It displays as
```
test`                                                                                                                              
test|                                                                                                                              
test*                                                                                                                                    
```

I am able to reproduce this on both the `nvim-macos.tar.gz` and `nvim-linux64.tar.gz`.

I build Neovim locally from source on MacOS and do not have this issue. So, it appears to be specific to the tarball build.

Additional context:
I run nightly (actually every 3 hour) tests on my plugin [kitty-scrollback.nvim](https://github.com/mikesmithgh/kitty-scrollback.nvim). This uses the github action [rhysd/action-setup-vim](https://github.com/rhysd/action-setup-vim) to download the nightly tarball of Neovim. 

Test failures started for me on Mar 18, 2024, 2:16 AM EDT.

See test failures [here](https://github.com/mikesmithgh/kitty-scrollback.nvim/actions/workflows/tests.yml?query=event%3Aschedule).

### Steps to reproduce

- `curl -L -o nvim.tar.gz https://github.com/neovim/neovim/releases/download/nightly/nvim-macos.tar.gz`
- `tar xvfz nvim.tar.gz` 
- `cd nvim-macos/`
- `VIMRUNTIME= ./bin/nvim --clean`
- insert the text `|hi|`
- run `:set filetype=help`

Results in the visible text:
```
hi|
```

### Expected behavior

Expected the visible text:
```
hi
```

### Neovim version (nvim -v)

NVIM v0.10.0-dev-2637+g400ef8aaa

### Vim (not Nvim) behaves the same?

n/a

### Operating system/version

macOS Version 14.2.1 (23C71)

### Terminal name/version

kitty 0.33.0

### $TERM environment variable

xterm-kitty

### Installation

Tarball linux or macos nightly build

**Comment by zeertzjq** [comment]
Fixed by #27934

**Comment by mikesmithgh** [comment]
confirmed it is fixed 👍
